File: 2.2/C++/Line.cpp
#include "Line.h"

Line::Line() {
    value = "#DEFAULT";
}

Line::Line(const std::string& input) {
    value = input;
}

std::string Line::GetValue() const {
    return value;
}

size_t Line::GetLength() const {
    return value.length();
}

void Line::Replace() {

}

File: 2.2/C#/Line.cs
using System;

namespace Lab
{
    public class Line 
    {
        protected string value;

         public Line()
        {
            value = "#DEFAULT";
        }

        public Line(string input)
        {
            value = input;
        }

        public virtual string GetValue()
        {
            return value;
        }

        public virtual int GetLength()
        {
            return value.Length;
        }

        public virtual void Replace() 
        { 

        } 
    }
}